04-testing-configuration-logging/01-unit-tests/test/Validator.test.js

```diff
@@ -3,20 +3,205 @@ const expect = require('chai').expect;
 
 describe('testing-configuration-logging/unit-tests', () => {
   describe('Validator', () => {
-    it('валидатор проверяет строковые поля', () => {
+    it('validates all the fields for min and max', () => {
       const validator = new Validator({
         name: {
           type: 'string',
+          min: 3,
+          max: 20,
+        },
+        age: {
+          type: 'number',
           min: 10,
           max: 20,
         },
       });
 
-      const errors = validator.validate({ name: 'Lalala' });
+      const errors = validator.validate({name: '', age: 1});
+      expect(errors).to.have.length(2);
+      expect(errors[0]).to.have.property('field').and.to.be.equal('name');
+      expect(errors[0]).to.have.property('error').and.to.be.equal('too short, expect 3, got 0');
+      expect(errors[1]).to.have.property('field').and.to.be.equal('age');
+      expect(errors[1]).to.have.property('error').and.to.be.equal('too little, expect 10, got 1');
+    });
 
-      expect(errors).to.have.length(1);
+    it('validates all the fields for the correct type', () => {
+      const validator = new Validator({
+        name: {
+          type: 'string',
+          min: 3,
+          max: 20,
+        },
+        age: {
+          type: 'number',
+          min: 10,
+          max: 20,
+        },
+      });
+
+      const errors = validator.validate({name: 1, age: ''});
+      expect(errors).to.have.length(2);
       expect(errors[0]).to.have.property('field').and.to.be.equal('name');
-      expect(errors[0]).to.have.property('error').and.to.be.equal('too short, expect 10, got 6');
+      expect(errors[0]).to.have.property('error').and.to.be.equal('expect string, got number');
+      expect(errors[1]).to.have.property('field').and.to.be.equal('age');
+      expect(errors[1]).to.have.property('error').and.to.be.equal('expect number, got string');
+    });
+
+    it('gives 0 errors for empty object', () => {
+      const validator = new Validator({
+        name: {
+          type: 'string',
+          min: 3,
+          max: 20,
+        },
+        age: {
+          type: 'number',
+          min: 10,
+          max: 20,
+        },
+      });
+
+      const errors = validator.validate({});
+      expect(errors).to.have.length(0);
+    });
+
+    it('ignores the field if it does not have a rule decribed', () => {
```

Comment:
Not sure if it is the desired implementation. Should we fail if object comes with the field we have no rules for?